File: old/Double Cola/1.py
Name =  ['Sheldon', 'Leonard', 'Penny', 'Rajesh', 'Howard']

x = int(raw_input())
y = 5
while True:
	if x<=y:
		y/=5
		print Name[(x-1)/y]
		break
	else:
		x-=y
		y*=2

File: old/Double Cola/1.java
import java.util.Scanner;
class doublecola{
	public static void main(String[] arg){
		Scanner in = new Scanner(System.in);
		int x,y=5;
		x = in.nextInt();
		String[] Name= new String[] {"Sheldon","Leonard","Penny","Rajesh","Howard"};
		while(true){
			if(x<=y){
				y/=5;
				System.out.print(Name[(x-1)/y]);
				break;
			}
			else{
				x-=y;
				y*=2;
			}
		}

	}
}